Disable MQTTsetMQTT and MQTT_HTTPS_FW_UPDATE to allow OTA for Avatto

DigiH opened the pull request:
Trying to update my Avattos with the currently released 0.9.8 I found that OTA wasn't possible any longer as the build was now too large

0.9.8
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM: [==== ] 42.4% (used 34772 bytes from 81920 bytes)
Flash: [====== ] 55.2% (used 528716 bytes from 958448 bytes)

this goes back to 0.9.7, where WiFi, MQTT and firmware update was introduced over MQTT commands

0.9.7
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM: [==== ] 42.4% (used 34772 bytes from 81920 bytes)
Flash: [====== ] 55.2% (used 528716 bytes from 958448 bytes)

Going back to building 0.9.6 I was able to OTA update again with the much smaller build

0.9.6
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM: [==== ] 39.7% (used 32512 bytes from 81920 bytes)
Flash: [==== ] 41.4% (used 396804 bytes from 958448 bytes)

Building 0.9.8 with the flags brings the build back to OTAable size

0.9.8 with
'-UMQTTsetMQTT'
'-UMQTT_HTTPS_FW_UPDATE'
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM: [==== ] 39.9% (used 32684 bytes from 81920 bytes)
Flash: [==== ] 42.3% (used 405024 bytes from 958448 bytes)

Similar to the flag setting in [env:rfbridge]

I'm not even sure if MQTT_HTTPS_FW_UPDATE would even work on the Avatto, with the limited flash memory.

Should we also add

board_build.ldscript = eagle.flash.1m64.ld

as a default, as it's required for OTA updates on the Avattos as discussed in 1technophile#763 (comment) ?

DigiH committed:
Disable MQTTsetMQTT and MQTT_HTTPS_FW_UPDATE to allow OTA for Avatto Bakery IR

Trying to update my Avattos with the currently released 0.9.8 I found that OTA wasn't possible any longer as the build was now to large

0.9.8
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [====      ]  42.4% (used 34772 bytes from 81920 bytes)
Flash: [======    ]  55.2% (used 528716 bytes from 958448 bytes)

this goes back to 0.9.7, where WiFi, MQTT and firmware update was introduced over MQTT commands

0.9.7
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [====      ]  42.4% (used 34772 bytes from 81920 bytes)
Flash: [======    ]  55.2% (used 528716 bytes from 958448 bytes)

Going back to building 0.9.6 I was able to OTA update again with the much smaller build

0.9.6
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [====      ]  39.7% (used 32512 bytes from 81920 bytes)
Flash: [====      ]  41.4% (used 396804 bytes from 958448 bytes)

Building 0.9.8 with the flags brings the build back to OTAable size

0.9.8 with
	'-UMQTTsetMQTT'
	'-UMQTT_HTTPS_FW_UPDATE'
Checking size .pio/build/avatto-bakeey-ir/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [====      ]  39.9% (used 32684 bytes from 81920 bytes)
Flash: [====      ]  42.3% (used 405024 bytes from 958448 bytes)

Similar to the flag setting in [env:rfbridge]

I'm not even sure if MQTT_HTTPS_FW_UPDATE would even work on the Avatto, with the limited flash memory. 

Should we also add

board_build.ldscript = eagle.flash.1m64.ld

as a default, as it's required for OTA updates on the Avattos as discussed in https://github.com/1technophile/OpenMQTTGateway/issues/763#issuecomment-698864712 ?

DigiH committed:
Merge pull request #1 from DigiH/Disable-MQTTsetMQTT-and-MQTT_HTTPS_FW_UPDATE-to-allow-OTA-for-Avatto-Bakery-IR

Disable MQTTsetMQTT and MQTT_HTTPS_FW_UPDATE to allow OTA for Avatto …

1technophile commented:
Thanks for pointing this

1technophile commented:
> as a default, as it's required for OTA updates on the Avattos as discussed in #763 (comment) ?

I think so, could you add into the [docs ](https://docs.openmqttgateway.com/use/gateway.html#switching-brokers-and-using-self-signed-and-client-certificates) that the 2 boards doesn't have these functions on the pre built binaries please

DigiH committed:
Doc addition for restricted MQTT options

Doc addition for https://github.com/1technophile/OpenMQTTGateway/pull/1046 and https://github.com/1technophile/OpenMQTTGateway/pull/986

DigiH committed:
Merge pull request #2 from DigiH/Doc-addition

Doc addition for restricted MQTT options

DigiH commented:
Added a warning in the docs.

1technophile commented:
Thanks
